### PR TITLE
bugfix: generate sandboxId to be stored advanced

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

If we rely on the id which returned by pouchd, we must store the id of sandbox after sandboxContainer has been created, if the process exits between creating container completely and storing it, the container will not be controlled by sandboxstore. So we generate the id of sandbox container and use it to create the container. 

For generating the container id, I use the interface of  `ContainerMgr.Get` to determine if the id already exists.

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

None.

### CODE REVIEW

**Comment by HusterWan** [comment]
please rebase your pr

File: cri/v1alpha2/cri.go [hunk 1]
```diff
@@ -244,34 +243,50 @@ func (c *CriManager) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
...
+	defer func() {
+		if retErr != nil && !removeContainerErr {
+			if err := c.SandboxStore.Remove(id); err != nil {
+				logrus.Errorf("failed to remove the metadata of container %q from sandboxStore: %v", id, err)
+			}
+		}
+	}()
+
+	// applies the annotations extended.
+	if err := c.applySandboxAnnotations(sandboxMeta, config.Annotations); err != nil {
```
[the remark this answers is not in the document]
Comment (Starnop) [reply cri/v1alpha2/cri.go]: It's just a code structure adjustment, not a new feature. So I think it's not necessary to be splited.  ^_^

File: cri/v1alpha2/cri.go [hunk 2]
```diff
@@ -244,34 +243,50 @@ func (c *CriManager) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	}
 
 	// Step 2: Create the sandbox container.
+
```
Comment (HusterWan) [inline cri/v1alpha2/cri.go]: delete this blank line ?

File: cri/v1alpha2/cri.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (HusterWan) [reply cri/v1alpha2/cri.go]: got you

File: cri/v1alpha2/cri.go (hunk 2 above)
Comment (Starnop) [reply cri/v1alpha2/cri.go]: This comment `Step2` represents it's a section down here , and should be separated from the comments below by a blank line. So I suggest keeping it. ^_^

**Comment by Starnop** [comment]
> please rebase your pr

Done.

File: cri/v1alpha2/cri_utils.go [hunk 3]
```diff
@@ -246,6 +264,31 @@ func applySandboxLinuxOptions(hc *apitypes.HostConfig, lc *runtime.LinuxPodSandb
...
+	if _, ok := annotations[anno.KubernetesRuntime]; ok {
+		sandboxMeta.Runtime = annotations[anno.KubernetesRuntime]
+		if err := c.SandboxStore.Put(sandboxMeta); err != nil {
+			return err
+		}
+	}
+
+	// apply the annotation of io.kubernetes.lxcfs.enabled
+	// which specify whether to enable lxcfs for a container.
+	if _, ok := annotations[anno.LxcfsEnabled]; ok {
```
[the remark this answers is not in the document]
Comment (Starnop) [reply cri/v1alpha2/cri_utils.go]: Done. ^_^

**Comment by HusterWan** [comment]
LGTM

**Pull request merged**